### PR TITLE
WTH template updates

### PR DESCRIPTION
**Pull request by jrzyshr**
This is a set of updates to the WTH Markdown templates in the 000-HowToHack folder.

These updates incorporate comments & sample content that surface many of the rules from the WTH Author guide right in the template for when a contributor is developing a new hack.  The updates also have "hidden" scripting tags that work with the "Create New Hack" GitHub Action that can scaffold out a new hack.

### CODE REVIEW

**Review by Whowong — APPROVED** [review]

File: 000-HowToHack/WTH-Challenge-Template.md [hunk 1]
```diff
@@ -1,82 +1,109 @@
...
 
-**- Fusce commodo nulla elit, vitae scelerisque lorem maximus eu.** 
+*Optionally, the coach or event host is encouraged to present a mini-lesson (with a PPT or video) to set up the context & introduction to each challenge. A summary of the content of that mini-lesson is a good candidate for this Introduction section*
 
-**- Nulla vitae ante turpis. Etiam tincidunt venenatis mauris, ac volutpat augue rutrum sed. Vivamus dignissim est sed dolor luctus aliquet. Vestibulum cursus turpis nec finibus commodo.**
+*For example:*
 
-**- Vivamus venenatis accumsan neque non lacinia. Sed maximus sodales varius. Proin eu nulla nunc. Proin scelerisque ipsum in massa tincidunt venenatis. Nulla eget interdum nunc, in vehicula risus.**
+When setting up an IoT device, it is important to understand how 'thingamajigs' work. Thingamajigs are a key part of every IoT device and ensure they are able to communicate properly with edge servers. Thingamajigs  require IP addresses to be assigned to them by a server and thus must have unique MAC addresses. In this challenge, you will get hands on with a thingamajig and learn how one is configured.
```
Comment (perktime) [inline 000-HowToHack/WTH-Challenge-Template.md:28]: Remove extra space between "Thingamajigs" and "require"

File: 000-HowToHack/WTH-ChallengeZero-Template.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+<!-- REMOVE_ME ${navigationLine} (remove this from your MD files if you are writing them manually, this is for the automation script) REMOVE_ME -->
+
+<!-- REPLACE_ME (this section will be removed by the automation script) -->
+<!-- If you are using and editing this template manually, ensure the navigation link below is updated to link to next challenge relative to the current challenge. The "Home" link should always link to the homepage of the hack which is the README.md in the hack's parent directory. -->
+**[Home](../README.md)** - [Next Challenge >](./Challenge-01.md)
+<!-- REPLACE_ME (this section will be removed by the automation script) -->
+
+***This is a template for "Challenge Zero" which focuses on getting prerequisites set up for the hack. The italicized text provides hints & examples of what should or should NOT go in each section.***
+
+***We have included links to some common What The Hack pre-reqs in this template. All common prerequisite links go to the WTH-CommonPrerequisites page where the are more details on what each tool's purpose is.***
```
Comment (perktime) [inline 000-HowToHack/WTH-ChallengeZero-Template.md:12]: should say "where there are more details..."

File: 000-HowToHack/WTH-ChallengeZero-Template.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    - [Note for Windows Users](../../000-HowToHack/WTH-Common-Prerequisites.md#note-for-windows-users)
+    - [Azure PowerShell CmdLets](../../000-HowToHack/WTH-Common-Prerequisites.md#azure-powershell-cmdlets)
+  - [Azure Cloud Shell](../../000-HowToHack/WTH-Common-Prerequisites.md#azure-cloud-shell)
+- [Visual Studio Code](../../000-HowToHack/WTH-Common-Prerequisites.md#visual-studio-code)
+  - [VS Code plugin for ARM Templates](../../000-HowToHack/WTH-Common-Prerequisites.md#visual-studio-code-plugins-for-arm-templates)
+- [Azure Storage Explorer](../../000-HowToHack/WTH-Common-Prerequisites.md#azure-storage-explorer)
+
+## Description
+
+*This section should clearly state any additional prerequisite tools that need to be installed, or set up in the Azure environment that the student will hack in.*
```
Comment (perktime) [inline 000-HowToHack/WTH-ChallengeZero-Template.md:48]: Remove comma

File: 000-HowToHack/WTH-ChallengeZero-Template.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+*While ordered lists are generally not welcome in What The Hack challenge descriptions, you can use one here in Challenge Zero IF and only IF the steps you are asking the student to perform are not core to the learning objectives of the hack.*
+
+*For example, if the hack is on IoT Devices and you want the student to deploy an ARM/Bicep template that sets up the environment they will hack in without them needing to understand how ARM/Bicep templates work, you can provide step-by-step instructions on how to deploy the ARM/Bicep template.*
+
+*Optionally, you may provide resource files such as a sample application, code snippets, or templates as learning aids for the students. These files are stored in the hack's \`Student/Resources\` folder. It is the coach's responsibility to package these resources into a Resources.zip file and provide it to the students at the start of the hack. You should leave the sample text below in that refers to the Resources.zip file.*
+
+***NOTE:** Do NOT provide direct links to files or folders in the What The Hack repository from the student guide. Instead, you should refer to the Resource.zip file provided by the coach.*
```
Comment (perktime) [inline 000-HowToHack/WTH-ChallengeZero-Template.md:56]: Should be Resources.zip (plural)

File: 000-HowToHack/WTH-ChallengeZero-Template.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+*Sample challenge zero text for the IoT Hack Of The Century:*
+
+Now that you have the common pre-requisites installed on your workstation, there are prerequisites specifc to this hack.
+
+Your coach will provide you with a Resources.zip file that contains resources you will need to complete the hack. If you plan to work locally, you should unpack it on your workstation. If you plan to use the Azure Cloud Shell, you should upload it to the Cloud Shell and unpack it there.
+
+Please install these additional tools:
+- [Azure IoT Tools](https://marketplace.visualstudio.com/items?itemName=vsciot-vscode.azure-iot-tools) extension for Visual Studio Code
+- .NET Core SDK 2.1.0 or later installed on your development machine. This can be downloaded from [here](https://www.microsoft.com/net/download/all) for multiple platforms.
```
Comment (perktime) [inline 000-HowToHack/WTH-ChallengeZero-Template.md:68]: .NET core SDK 2.1.0 reached end of support in August 2021 so we mention a later version than this one.

**Review by perktime — CHANGES_REQUESTED** [review]

File: 000-HowToHack/WTH-Challenge-Template.md (hunk 1 above)
Comment (jrzyshr) [reply 000-HowToHack/WTH-Challenge-Template.md:28]: Done.

File: 000-HowToHack/WTH-ChallengeZero-Template.md (hunk 2 above)
Comment (jrzyshr) [reply 000-HowToHack/WTH-ChallengeZero-Template.md:12]: done.

File: 000-HowToHack/WTH-ChallengeZero-Template.md (hunk 3 above)
Comment (jrzyshr) [reply 000-HowToHack/WTH-ChallengeZero-Template.md:48]: Done

File: 000-HowToHack/WTH-ChallengeZero-Template.md (hunk 4 above)
Comment (jrzyshr) [reply 000-HowToHack/WTH-ChallengeZero-Template.md:56]: Done.

File: 000-HowToHack/WTH-ChallengeZero-Template.md (hunk 5 above)
Comment (jrzyshr) [reply 000-HowToHack/WTH-ChallengeZero-Template.md:68]: This is an example pre-req.  It's a copy/paste of the actual pre-reqs from our real IoT Hack... 029-IoTEdge!  Maybe that hack should be updated too.  (I bumped it to 6.0 for the sample.)

Done.

**Review by jordanbean-msft — APPROVED** [review]

**Pull request merged**